src/attr/__init__.pyi

```diff
@@ -41,6 +42,10 @@ _ConverterType = Callable[[Any], _T]
 _FilterType = Callable[[Attribute[_T], _T], bool]
 _ReprType = Callable[[Any], str]
 _ReprArgType = Union[bool, _ReprType]
+_OnSetAttrType = Callable[[Any, Any, Any], Any]
```

Comment:
Could this be:
```
Callable[[Any, Attribute[_T], _T], _T]
```
?
It might just be enough to do:
```
Callable[[Any, Attribute[Any], Any], Any]
```

Reply:
What would those `_T`s buy us?

Reply:
The _T means that everywhere you see it, the type must be the same.  So mypy would warn you if your method tries to change the type or if the attribute doesn't match the new value.  For example

```
def foo(instance: Any, attribute: Attribute[bool], new_value: str) -> bool:
    return new_value == 'true'

@attr.s(on_setattr=foo)
```

However looking at it, this might be something you want to do.  So I think the correct type is
```
Callable[[Any, Attribute[_T], Any], _T]
```

Basically it forces your on_setattr method to return a type that's the same as the Attribute.

Reply:
Is this something we want to enforce? 🤔 If you have a list, you could have intermediate results and only the last one has to return the correct one.

Reply:
Oh wow.  Hadn't thought of that.  Can converters be chained in the same way?

Reply:
They will be with the next release. At this point it's called `chain()` but I think `pipe()` is better and I will rename it before I release it.

Alternatively I could be convinced to rename pipe to chain here, but I feel like pipe is better at conveying the fact, that the data flows thru it?

Reply:
I went with `Attribute[Any]` for now